cms_account_form/models/account_form.py

```diff
@@ -0,0 +1,157 @@
+        _fields['email']['help'] = tmpl.render({})
+
+    def form_validate_email(self, value, **req_values):
+        error, message = None, None
+        if value and not validate_email(value):
+            error = 'email_not_valid'
+            message = _('Invalid Email! Please enter a valid email address.')
+        return error, message
+
+    def form_validate_vat(self, value, **req_values):
+        error, message = None, None
+        if value and hasattr(self.form_model, 'check_vat'):
+            country_id = int(req_values.get("country_id", 0))
+            if country_id:
+                value = self.form_model.fix_eu_vat_number(country_id, value)
+            partner_dummy = self.form_model.new({
+                'vat': value,
+                'country_id': country_id or False,
+            })
+            try:
+                partner_dummy.check_vat()
+            except exceptions.ValidationError as err:
+                error = 'vat_not_valid'
+                message = err.name
+        return error, message
+
+    def form_before_create_or_update(self, values, extra_values):
+        user = self.env.user
+        # handle email update
+        if 'email' in values and user.email != values.get('email'):
+            self._handle_email_update(user, values)
+
+    def _handle_email_update(self, user, values):
+        """Validate email update and handle login update.
+
+        :param user: res.users record
+        :param values: form request values
+
+        If email is validated:
+
+            * check if already exists
+            * if yes, provide an error message
+            * if not, updates user's login
+            * provide an info message
+            * logout current session to force re-login with new email
+        """
+        email = values['email']
+        # TODO: this could be useless as we validate w/ `form_validate_email`
+        valid = validate_email(email)
+        if email and valid and not user._is_admin():
+            exists = user.sudo().search_count(
+                ['|', ('login', '=', email), ('email', '=', email)]
+            )
+            if exists and self.o_request.website:
+                # prevent email save and display friendly message
+                del values['email']
+                self._handle_email_exists(email)
+                return False
+            try:
+                self._handle_login_update(email, user)
+                can_change = True
+            except MailDeliveryException:
+                # do not update email / login
+                # if for any reason we cannot send email
+                can_change = False
+            if can_change and self.o_request.website:
+                self._logout_and_notify(email)
+            return True
+        return False
+
+    def _handle_email_exists(self, email):
+        title = _('Warning')
+        msg = _(
+            'Email address `%s` already taken. '
+            'Please check inside your company. '
```

Comment:
Is "Please check inside your company." helpful?

Reply:
bah, a legacy string from our project ;)